File: python-pr/52/python/code-quality-error-handling/network_client_32.py
# {fact rule=code-quality-error-handling@v1.0 defects=0}
def fetch_url(url):
    import urllib.request
    import urllib.error
    try:
        response = urllib.request.urlopen(url)
        return response.read()
    except urllib.error.URLError as e:
        print(f"URL error: {e}")
        return None
# {/fact}

File: js-pr/56/javascript/code-quality-error-handling/network_client_32.js
// {fact rule=code-quality-error-handling@v1.0 defects=0}
async function fetchData(url) {
    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        return await response.json();
    } catch (error) {
        console.error('Network error:', error.message);
        return null;
    }
}
// {/fact}